mutiny-core/src/storage.rs

```diff
@@ -137,6 +147,54 @@ pub trait MutinyStorage: Clone + Sized + 'static {
         self.set(key, json)
     }
 
+    /// Set a value in the storage, the function will encrypt the value if needed
+    async fn set_data_async<T>(
+        &self,
+        key: impl AsRef<str>,
+        value: T,
+        version: Option<u32>,
+    ) -> Result<(), MutinyError>
+    where
+        T: Serialize,
+    {
+        let data = serde_json::to_value(value).map_err(|e| MutinyError::PersistenceFailed {
+            source: MutinyStorageError::SerdeError { source: e },
+        })?;
+
+        let json: Value = encrypt_value(key.as_ref(), data.clone(), self.cipher())?;
+        let local_fut = self.set_async(key.as_ref(), json);
+
+        let vss_fut = async {
+            if let (Some(vss), Some(version)) = (self.vss_client(), version) {
+                let item = VssKeyValueItem {
+                    key: key.as_ref().to_string(),
+                    value: data,
+                    version,
+                };
+
+                vss.put_objects(vec![item]).await
+            } else {
+                Ok(())
+            }
+        };
+
+        let (local_res, vss_res) = futures::join!(local_fut, vss_fut);
+
+        // make sure local storage succeeded
+        local_res?;
+
+        // log error if VSS failed, if we have local version it should be okay
```

Comment:
It might be, though unless the user is doing multi device.. 

If they go to another device then it needs to save to the cloud as well.

Reply:
We might have to go about multi device in a different way in general. IE, safe to do unless we can tell that there's an out of date cloud state. In which case, we need to make sure a force close doesn't happen and then the user can go back to the original device or go to the emergency flow to redeem things on chain in a SCB kind of way.